controllers/secret_watcher.go

```diff
@@ -0,0 +1,102 @@
+package controllers
+
+import (
+	"context"
+	"time"
+
+	constants "github.com/cybozu-go/meows"
+	meowsv1alpha1 "github.com/cybozu-go/meows/api/v1alpha1"
+	"github.com/cybozu-go/meows/github"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+func NewSecretWatcher(client client.Client, interval time.Duration, githubClient github.Client) SecretWatcher {
+	return SecretWatcher{
+		client:       client,
+		interval:     interval,
+		githubClient: githubClient,
+	}
+}
+
+type SecretWatcher struct {
+	client       client.Client
+	interval     time.Duration
+	githubClient github.Client
+}
+
+// Start implements Runnable.Start
+func (w SecretWatcher) Start(ctx context.Context) error {
+	ticker := time.NewTicker(w.interval)
+	defer ticker.Stop()
+	logger := log.FromContext(ctx).WithName("SecretWatcher")
+	for {
+		select {
+		case <-ctx.Done():
+			return ctx.Err()
+		case <-ticker.C:
+			logger.Info("start secret watcher")
+			err := w.secretExpired(ctx)
+			if err != nil {
+				return err
+			}
+		}
+	}
+}
+
+func (w SecretWatcher) secretExpired(ctx context.Context) error {
+	logger := log.FromContext(ctx).WithName("SecretWatcher")
+	rps := meowsv1alpha1.RunnerPoolList{}
+	err := w.client.List(ctx, &rps)
+	if err != nil {
+		return err
+	}
+	for i := range rps.Items {
+		rp := &rps.Items[i]
+		s := new(corev1.Secret)
+		err := w.client.Get(ctx, types.NamespacedName{Name: rp.GetRunnerSecretName(), Namespace: rp.Namespace}, s)
```

Comment:
You can use `FieldIndexer` to efficiently list secrets owned by RunnerPool.

Reply:
I understood that It could use the `FieldIndexer` to efficiently list secrets owned by RunnerPool.
However, it did not seem to work well in this case because it required `.spec.repositoryName` in the RunnerPool to issue the token.